e2e/tests/functional/recentObjects.e2e.spec.js

```diff
@@ -102,31 +101,117 @@ test.describe('Recent Objects', () => {
+
+        // Wait for highlight animation to end
+        await waitForAnimations(clockTreeItem.locator('.c-tree__item'));
+
+        // Assert that the Clock treeitem is no longer highlighted
+        await expect(clockTreeItem.locator('.c-tree__item')).not.toHaveClass(/is-targeted-item/);
     });
-    test.fixme("Tests for context menu actions from recent objects", async ({ page }) => {
+    test("Persists on refresh", async ({ page }) => {
+        assertInitialRecentObjectsListState();
+        await page.reload();
+        assertInitialRecentObjectsListState();
     });
+    test("Displays objects and aliases uniquely", async ({ page }) => {
+        const mainTree = page.getByRole('tree', { name: 'Main Tree'});
+
+        // Navigate to the clock and reveal it in the tree
+        await page.goto(clock.url);
+        await page.getByTitle('Show selected item in tree').click();
+
+        // Right click the clock and create an alias using the "link" context menu action
+        const clockTreeItem = page.getByRole('tree', {
+            name: 'Main Tree'
+        }).getByRole('treeitem', {
+            name: clock.name
+        });
+        await clockTreeItem.click({
+            button: 'right'
+        });
+        await page.getByRole('menuitem', {
+            name: /Create Link/
+        }).click();
+        await page.getByRole('tree', { name: 'Create Modal Tree'}).getByRole('treeitem').first().click();
+        await page.getByRole('button', {
+            name: 'Save'
+        }).click();
+
+        // Click the newly created object alias in the tree
+        await mainTree.getByRole('treeitem', {
+            name: new RegExp(clock.name)
+        }).filter({
+            has: page.locator('.is-alias')
+        }).click();
+
+        // Assert that two recent objects are displayed and one of them is an alias
+        expect(await recentObjectsList.getByRole('listitem', { name: clock.name }).count()).toBe(2);
+        expect(await recentObjectsList.locator('.is-alias').count()).toBe(1);
+
+        // Assert that the alias and the original's breadcrumbs are different
+        const clockBreadcrumbs = recentObjectsList.getByRole('listitem', {name: clock.name}).getByRole('navigation');
+        expect(await clockBreadcrumbs.count()).toBe(2);
+        expect(await clockBreadcrumbs.nth(0).innerText()).not.toEqual(await clockBreadcrumbs.nth(1).innerText());
+    });
+    test("Enforces a limit of 20 recent objects", async ({ page }) => {
+        // Creating 21 objects takes a while, so increase the timeout
+        test.slow();
+
+        // Create 19 more objects (2 in beforeEach() + 19 new = 21 total)
+        for (let i = 0; i < 19; i++) {
+            await createDomainObjectWithDefaults(page, {
+                type: "Clock"
+            });
+        }
+
+        // Assert that the list contains 20 objects
+        expect(await recentObjectsList.locator('.c-recentobjects-listitem').count()).toBe(20);
```

Comment:
is there anything else worth asserting at this point given the test execution time?

Reply:
Updated the test to do another test for the "target" button, this time on an extremely deeply nested object